src/main/java/io/vertx/core/spi/loadbalancing/PreferenceAwareEndpointSelector.java

```diff
@@ -0,0 +1,16 @@
+package io.vertx.core.spi.loadbalancing;
+
+import java.util.List;
+
+public interface PreferenceAwareEndpointSelector extends EndpointSelectorCommons {
+
+  /**
+   * Select an endpoint among the provided list taking an attribute into account.
+   *
+   * @param endpoints the endpoint
+   * @param attribute the attribute TODO
```

Comment:
I think we can do better for the naming/types?
also for the `PreferenceAwarexxx`